cip/CIP2016-06-14-Define-comparability-and-equality-as-well-as-orderability-and-equivalence.adoc

```diff
@@ -0,0 +1,377 @@
+  * Positive infinity is of type `FLOAT`, equal to itself and greater than any other number (excluding `NaN`).
+  * Negative infinity is of type `FLOAT`, equal to itself and less than any other number (excluding `NaN`).
+  * `NaN` is <<incomparable>>.
+  * Numbers are <<incomparable>> to any value that is not also a number.
+- Booleans
+  * Booleans are compared such that `false` is less than `true`.
+  * Booleans are <<incomparable>> to any value that is not also a boolean.
+- Strings
+  * Strings are compared in dictionary order, i.e. characters are compared pairwise in ascending order from the start of the string to the end.
+  Characters missing in a shorter string are considered to be less than any other character.
+  For example, `'a' < 'aa'`.
+  * Strings are <<incomparable>> to any value that is not also a string.
+- Lists
+  * Lists are compared in dictionary order, i.e. list elements are compared pairwise in ascending order from the start of the list to the end. Elements missing in a shorter list are considered to be less than any other value.
+  For example, `[1] < [1, 0]`.
+  * Lists containing `null` values are <<incomparable>>. For example, `[1] >= [1, null]` evaluates to `null`.
+  * Lists are <<incomparable>> to any value that is not also a list.
+- Maps
+  * [[regular-maps,regular maps]]Regular maps
+  ** The comparison order for maps is unspecified and left to implementations.
+  ** The comparison order for maps must align with the <<equality-def,equality semantics>> outlined below.
+  In consequence, any map that contains an entry that maps its key to a `null` value is <<incomparable>>.
+  For exampe, `{a: 1} \<= {a: 1, b: null}` evaluates to `null`.
+  ** Regular maps are <<incomparable>> to any value that is not also a regular map.
+  * Nodes
+  ** The comparison order for nodes is based on an implementation specific internal total order of node identities.
+  ** Nodes are <<incomparable>> to any value that is not also a node.
+  * Relationships
+  ** The comparison order for relationships is based on an implementation specific internal total order of relationship identities.
+  ** Relationships are <<incomparable>> to any value that is not also a relationship.
+- Paths
+  ** Paths are compared as if they were a list of alternating nodes and relationships of the path from the start node to the end node.
+  For example, given nodes `n1`, `n2`, `n3`, and relationships `r1` and `r3`, and given that `n1 < n2 < n3` and `r1 < r3`, then the path `p1` from `n1` to `n3` via `r1` would be less than the path `p2` to `n1` from `n2` via `r3`. Expressed in terms of lists:
+
+  p1 < p2
+  => [n1-r1->n3] < [n1<-r3-n2]
+  => n1 < n1 || r1 < r3 || n3 < n2
+  => false || true || false
+  => true
```

Comment:
this example does not illustrate the actual comparison, since it deals only with `true` and `false` and _or_. Actual ordering comparison would have to deal with _less than_, _greater than_, and _equal_, and combining that.